src/Transformer/UserTransformer.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Transformer;
+
+use App\DTO\UserDTO;
+use App\Entity\User;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+
+class UserTransformer
+{
+    /**
+     * @var UserPasswordEncoderInterface
+     */
+    private $passwordEncoder;
+
+    public function __construct(UserPasswordEncoderInterface $passwordEncoder)
+    {
+        $this->passwordEncoder = $passwordEncoder;
+    }
+
+    public function transformDTOToEntity(UserDTO $dto, ?User $user = null): User
+    {
+        if ($user === null) {
+            $user = new User();
+            $user->setUsername($dto->username);
+            $user->setEmail($dto->email);
+            $user->setPassword($dto->password);
```

Comment:
Why you cannot change the password for an existing user? And why are you setting the password unencrypted?